### ISSUE TITLE
While training GAN using Keras 3.0 + JAX, loss doesn't seems right

### ISSUE THREAD

**Issue by sineeli** [keras-team/keras#18914]
Hi,

I tried to construct this network with the help of [Keras tutorial custom training step using JAX](https://keras.io/guides/writing_a_custom_training_loop_in_jax/)

Probably I made a mistake in Generator Training Step where loss and gradients are computed. Please check below code for generator training step. I have tired best to solve but not able find the mistake.

```python3
def generator_train_step(batch_size,
                         latent_dim,
                         gen_state):
    """Trains the generator.

    Args:
      trainable_variables: List of trainable variables.
      non_trainable_variables: List of non-trainable variables.
      z: Batch of noise of shape (batch_size, latent_dim).
      labels: Batch of labels of shape (batch_size, 1).
      
    Returns:
      loss: float, loss value.
      state: Tuple of (trainable_variables, optimizer_variables, state).
    """

    def compute_loss_and_updates(
        gen_trainable_variables, 
        gen_non_trainable_variables, 
        disc_trainable_variables, 
        disc_non_trainable_variables,
        z,
        labels):

        # Generate images from noise.
        generated_images, non_trainable_variables = vanilla_generator.stateless_call(
            gen_trainable_variables, gen_non_trainable_variables, z
        )

        # Discriminator predictions for the fake images
        predictions, _ =  vanilla_discriminator.stateless_call(
            disc_trainable_variables, disc_non_trainable_variables, generated_images)

        # Generator tries to fool discriminator, so it wants to minimize -log(D(G(z)))
        loss = loss_fn(labels, predictions)
        return loss, non_trainable_variables
    

    # Generate noise.
    z = np.random.normal(size=(batch_size, latent_dim))


    labels = jnp.ones(shape=(batch_size, 1))

    # Unpack the generator and discriminator states
    gen_trainable_variables, gen_non_trainable_variables, gen_optimizer_variables = gen_state
    disc_trainable_variables, disc_non_trainable_variables, disc_optimizer_variables = disc_state

    # Compute loss and gradients
    grad_fn = jax.value_and_grad(compute_loss_and_updates, has_aux=True)
    (loss, gen_non_trainable_variables), grads = grad_fn(
        gen_trainable_variables, 
        gen_non_trainable_variables, 
        disc_trainable_variables, 
        disc_non_trainable_variables,
        z, 
        labels)
    # Update generator parameters
    gen_trainable_variables, gen_optimizer_variables = generator_optimizer.stateless_apply(
        gen_optimizer_variables, grads, gen_trainable_variables
    )

    # Return the loss and updated state
    return loss, (
        gen_trainable_variables,
        gen_non_trainable_variables,
        gen_optimizer_variables,
    )

```

Also the whole gist of the notebooks is [here](https://colab.sandbox.google.com/gist/sineeli/2e21d6465085a6e9c89cc561fda22a9e/gan-keras-3-0-jax.ipynb), thanks in advance for the help.

**Comment by divyashreepathihalli** [comment]
@sineeli we will not e debugging users code. If you do find an issue with the framework, please open an issue and provide a repo. Thank you!
